iaso/tests/api/test_group_sets.py

```diff
@@ -0,0 +1,373 @@
+
+        cls.acccount_1_user_1 = cls.create_user_with_profile(
+            username="yoda", account=star_wars, permissions=["iaso_org_units"]
+        )
+        cls.acccount_1_user_2 = cls.create_user_with_profile(username="chewbacca", account=star_wars)
+        cls.acccount_3_user_1 = cls.create_user_with_profile(
+            username="raccoon", account=marvel, permissions=["iaso_org_units"]
+        )
+
+        cls.project_1 = m.Project.objects.create(
+            name="Hydroponic gardens", app_id="stars.empire.agriculture.hydroponics", account=star_wars
+        )
+
+        cls.project_2 = m.Project.objects.create(
+            name="New Land Speeder concept", app_id="stars.empire.agriculture.land_speeder", account=star_wars
+        )
+
+        cls.src_1_group_1 = m.Group.objects.create(name="Councils", source_version=cls.source_version_1)
+        cls.src_1_group_2 = m.Group.objects.create(name="Hidden", source_version=cls.source_version_1)
+
+        cls.src_2_group_1 = m.Group.objects.create(name="Assemblies", source_version=cls.source_version_2)
+
+        cls.project_1.data_sources.add(cls.data_source)
+        cls.project_1.save()
+        cls.data_source.account = star_wars
+        cls.data_source.default_version = cls.source_version_1
+        cls.data_source.projects.add(cls.project_1)
+        cls.data_source.save()
+
+        cls.data_source_2.account = star_wars
+        cls.data_source_2.projects.add(cls.project_1)
+        cls.data_source_2.save()
+
+        cls.acccount_1_user_1.iaso_profile.projects.add(cls.project_1)
+
+    def test_authentification_required(self):
+        response = self.client.get("/api/group_sets/", format="json")
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_create_groupset_with_valid_groups(self):
+        """
+        Ensure we can create a GroupSet with valid group_ids.
+        """
+
+        self.client.force_authenticate(self.acccount_1_user_1)
+
+        valid_payload = {
+            "name": "New GroupSet",
+            "source_version_id": self.source_version_1.id,
+            "source_ref": "Bpx0589u8y0",
+            "group_ids": [self.src_1_group_1.id, self.src_1_group_2.id],
+        }
+        response = self.client.post("/api/group_sets/", valid_payload, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(GroupSet.objects.count(), 1)
+
+        created_groupset = GroupSet.objects.last()
+        self.assertEqual(created_groupset.name, "New GroupSet")
+        self.assertEqual(created_groupset.source_ref, valid_payload["source_ref"])
+        self.assertEqual(created_groupset.source_version.id, self.source_version_1.id)
+        self.assertEqual(created_groupset.group_belonging, GroupSet.GroupBelonging.SINGLE)
+        self.assertEqual([x.id for x in created_groupset.groups.all()], [self.src_1_group_1.id, self.src_1_group_2.id])
```

Comment:
Instead of calling `.all()` and doing list comprehension, I used in OUCRC `.values_list("id", flat=True)`. Not sure if that changes a lot since it's run on a test DB without a lot of data